vmm/src/lib.rs

```diff
@@ -255,16 +255,13 @@ impl Vmm {
         // Without ACPI, a reset is equivalent to a shutdown
         #[cfg(not(feature = "acpi"))]
         {
-            if let Some(ref mut vm) = self.vm {
-                vm.shutdown()?;
-                return Ok(());
-            }
+            self.vm_shutdown()?;
```

Comment:
You still want to `return Ok(());` here as non ACPI platforms will not be able to reset after a shutdown.

Reply:
Or rather, this should be:

```patch
diff --git a/vmm/src/lib.rs b/vmm/src/lib.rs
index 75f3653..802bd4d 100644
--- a/vmm/src/lib.rs
+++ b/vmm/src/lib.rs
@@ -255,10 +255,7 @@ impl Vmm {
         // Without ACPI, a reset is equivalent to a shutdown
         #[cfg(not(feature = "acpi"))]
         {
-            if let Some(ref mut vm) = self.vm {
-                vm.shutdown()?;
-                return Ok(());
-            }
+            return self.vm_shutdown();
         }
 
         // First we stop the current VM and create a new one.
```

Reply:
Oh, yeah, I ignore the return keyword, let me add it. The latter seems better.